File: DoConnect/src/main/java/com/wipro/doconnect/service/UserService.java
package com.wipro.doconnect.service;

import java.io.IOException;
import java.util.List;
import javax.validation.Valid;
import org.springframework.http.ResponseEntity.BodyBuilder;
import org.springframework.web.multipart.MultipartFile;
import com.wipro.doconnect.dto.AskQuestionDTO;
import com.wipro.doconnect.dto.Message;
import com.wipro.doconnect.dto.PostAnswerDTO;
import com.wipro.doconnect.dto.ResponseDTO;
import com.wipro.doconnect.entity.Answer;
import com.wipro.doconnect.entity.ImageModel;
import com.wipro.doconnect.entity.Question;
import com.wipro.doconnect.entity.User;
/*@Author:Arpit Verma
 * Modified Date: 27-08-2022
 * Description:User Service
 */
public interface UserService
{
	public User userLogin(String email, String password);
	public String userLogout(Long userId);
	public User userRegister(@Valid User user);
	
	public Question askQuestion(@Valid AskQuestionDTO askQuestionDTO);
	public Answer giveAnswer(@Valid PostAnswerDTO postAnswerDTO);
	
	
	public List<Question> searchQuestion(String question);
	
	public List<Answer> getAnswers(Long questionId);
	public List<Question> getQuestions(String topic);
	
	public BodyBuilder uplaodImage(MultipartFile file) throws IOException;
	public ImageModel getImage(String imageName);
	
	public Message sendMessage(@Valid Message message);
}

File: DoConnectUI/src/app/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { User } from './user';
import { Observable } from 'rxjs';
import { Question } from './question';
import { Answer } from './answer';


@Injectable({
  providedIn: 'root'
})
export class UserService {

  constructor(private http:HttpClient) {

   }

questionId:number=0
   user = new User()
  url:string="http://localhost:9090/user"

  sendUserData(user:User){
    this.user.id=user.id
	this.user.name=user.name
  this.user.email=user.email
  }

  giveUserData(){
    return this.user
   }

   getQuestionId(questionId:number){
    this.questionId=questionId
   }
   sendQuestionId(){
    return this.questionId

   }

	userLogin(email:string, password:string):Observable<User> {
		return this.http.get<User>(this.url+"/login/"+email+"/"+password);
	}

	userLogout( userId:number) {
		return this.http.get(this.url+"/logout/"+userId);
	}

	userRegister(user:User):Observable<User> {
		return this.http.post<User>(this.url+"/register",user)
	}

  askQuestion(askQuestionDTO:any):Observable<Question> {
	  return this.http.post<Question>(this.url+"/askQuestion",askQuestionDTO);
	}

  giveAnswer(postAnswerDTO:any):Observable<Answer> {
	  return this.http.post<Answer>(this.url+"/giveAnswer",postAnswerDTO);
	}

	searchQuestion(question:string):Observable<Question[]> {
	  return this.http.get<Question[]>(this.url+"/searchQuestion/"+question);
	}

  getAnswers(questionId:number):Observable<Answer[]> {
		return this.http.get<Answer[]>(this.url+"/getAnswers/"+questionId);
	}

	getQuestions(topic:string):Observable<Question[]> {
		return this.http.get<Question[]>(this.url+"/getQuestions/"+topic);
	}



}
